analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassNamedException.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+class CustomException { }                                        // Noncompliant {{Rename this class to remove "(e|E)xception" or correct its inheritance.}}
+//    ^^^^^^^^^^^^^^^
+class Customexception { }                                        // Noncompliant
+class CustomEXCEPTION { }                                        // Noncompliant
+
```

Comment:
If you consider it useful, you can add the following test (with generics).
```suggestion
class CustomException<T> { }                                     // Noncompliant
//    ^^^^^^^^^^^^^^^

```

Reply:
That's a really good point. I added test cases for generic classes.